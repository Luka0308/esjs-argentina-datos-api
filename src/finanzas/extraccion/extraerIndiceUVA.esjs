importar { load } desde 'cheerio'
importar { format, parse } desde 'date-fns'
importar { extraerBcra } desde '@/extractores/bcra.extractor.esjs'

exportar asincrono funcion extraerIndiceUVA(desde, hasta) {
  const html = esperar extraerBcra('7913', desde, hasta)

  const $ = load(html)

  const tabla = $('table')

  const items = []

  tabla.buscar('tbody tr').each((i, tr) => {
    const fecha = $(tr).buscar('td').eq(0).text()
    const valor = $(tr).buscar('td').eq(1).text()

    items.agregar({
      fecha: format(parse(fecha, 'dd/MM/yyyy', crear Fecha()), 'yyyy-MM-dd'),
      valor: Numero(valor.reemplazar(',', '.')),
    })
  })

  retornar items
}
