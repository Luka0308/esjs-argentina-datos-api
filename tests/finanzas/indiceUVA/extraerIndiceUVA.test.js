import { describe, expect, it } from 'vitest'
import { extraerIndiceUVA } from '@/finanzas/extraccion/extraerIndiceUVA.esjs'

describe('extraerIndiceUVA', () => {
  it('extrae los indices UVA', async () => {
    const items = await extraerIndiceUVA(
      '2023-01-01',
      '2024-01-01'
    )

    expect(items).toMatchSnapshot()
  })
})
