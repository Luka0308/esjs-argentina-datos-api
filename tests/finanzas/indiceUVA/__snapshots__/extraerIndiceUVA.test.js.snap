// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`extraerIndiceUVA > extrae los indices UVA 1`] = `
[
  {
    "fecha": "2023-01-01",
    "valor": 185.32,
  },
  {
    "fecha": "2023-01-02",
    "valor": 185.32,
  },
  {
    "fecha": "2023-01-03",
    "valor": 186.18,
  },
  {
    "fecha": "2023-01-04",
    "valor": 186.47,
  },
  {
    "fecha": "2023-01-05",
    "valor": 186.76,
  },
  {
    "fecha": "2023-01-06",
    "valor": 187.04,
  },
  {
    "fecha": "2023-01-07",
    "valor": 187.33,
  },
  {
    "fecha": "2023-01-08",
    "valor": 187.33,
  },
  {
    "fecha": "2023-01-09",
    "valor": 187.33,
  },
  {
    "fecha": "2023-01-10",
    "valor": 188.2,
  },
  {
    "fecha": "2023-01-11",
    "valor": 188.49,
  },
  {
    "fecha": "2023-01-12",
    "valor": 188.78,
  },
  {
    "fecha": "2023-01-13",
    "valor": 189.08,
  },
  {
    "fecha": "2023-01-14",
    "valor": 189.37,
  },
  {
    "fecha": "2023-01-15",
    "valor": 189.37,
  },
  {
    "fecha": "2023-01-16",
    "valor": 189.37,
  },
  {
    "fecha": "2023-01-17",
    "valor": 190.26,
  },
  {
    "fecha": "2023-01-18",
    "valor": 190.56,
  },
  {
    "fecha": "2023-01-19",
    "valor": 190.87,
  },
  {
    "fecha": "2023-01-20",
    "valor": 191.18,
  },
  {
    "fecha": "2023-01-21",
    "valor": 191.48,
  },
  {
    "fecha": "2023-01-22",
    "valor": 191.48,
  },
  {
    "fecha": "2023-01-23",
    "valor": 191.48,
  },
  {
    "fecha": "2023-01-24",
    "valor": 192.41,
  },
  {
    "fecha": "2023-01-25",
    "valor": 192.72,
  },
  {
    "fecha": "2023-01-26",
    "valor": 193.03,
  },
  {
    "fecha": "2023-01-27",
    "valor": 193.34,
  },
  {
    "fecha": "2023-01-28",
    "valor": 193.65,
  },
  {
    "fecha": "2023-01-29",
    "valor": 193.65,
  },
  {
    "fecha": "2023-01-30",
    "valor": 193.65,
  },
  {
    "fecha": "2023-01-31",
    "valor": 194.58,
  },
  {
    "fecha": "2023-02-01",
    "valor": 194.89,
  },
  {
    "fecha": "2023-02-02",
    "valor": 195.21,
  },
  {
    "fecha": "2023-02-03",
    "valor": 195.52,
  },
  {
    "fecha": "2023-02-04",
    "valor": 195.83,
  },
  {
    "fecha": "2023-02-05",
    "valor": 195.83,
  },
  {
    "fecha": "2023-02-06",
    "valor": 195.83,
  },
  {
    "fecha": "2023-02-07",
    "valor": 196.78,
  },
  {
    "fecha": "2023-02-08",
    "valor": 197.09,
  },
  {
    "fecha": "2023-02-09",
    "valor": 197.41,
  },
  {
    "fecha": "2023-02-10",
    "valor": 197.73,
  },
  {
    "fecha": "2023-02-11",
    "valor": 198.05,
  },
  {
    "fecha": "2023-02-12",
    "valor": 198.05,
  },
  {
    "fecha": "2023-02-13",
    "valor": 198.05,
  },
  {
    "fecha": "2023-02-14",
    "valor": 199,
  },
  {
    "fecha": "2023-02-15",
    "valor": 199.32,
  },
  {
    "fecha": "2023-02-16",
    "valor": 199.64,
  },
  {
    "fecha": "2023-02-17",
    "valor": 200.06,
  },
  {
    "fecha": "2023-02-18",
    "valor": 200.47,
  },
  {
    "fecha": "2023-02-19",
    "valor": 200.47,
  },
  {
    "fecha": "2023-02-20",
    "valor": 200.47,
  },
  {
    "fecha": "2023-02-21",
    "valor": 200.47,
  },
  {
    "fecha": "2023-02-22",
    "valor": 200.47,
  },
  {
    "fecha": "2023-02-23",
    "valor": 202.57,
  },
  {
    "fecha": "2023-02-24",
    "valor": 202.99,
  },
  {
    "fecha": "2023-02-25",
    "valor": 203.42,
  },
  {
    "fecha": "2023-02-26",
    "valor": 203.42,
  },
  {
    "fecha": "2023-02-27",
    "valor": 203.42,
  },
  {
    "fecha": "2023-02-28",
    "valor": 204.69,
  },
  {
    "fecha": "2023-03-01",
    "valor": 205.12,
  },
  {
    "fecha": "2023-03-02",
    "valor": 205.54,
  },
  {
    "fecha": "2023-03-03",
    "valor": 205.97,
  },
  {
    "fecha": "2023-03-04",
    "valor": 206.4,
  },
  {
    "fecha": "2023-03-05",
    "valor": 206.4,
  },
  {
    "fecha": "2023-03-06",
    "valor": 206.4,
  },
  {
    "fecha": "2023-03-07",
    "valor": 207.69,
  },
  {
    "fecha": "2023-03-08",
    "valor": 208.13,
  },
  {
    "fecha": "2023-03-09",
    "valor": 208.56,
  },
  {
    "fecha": "2023-03-10",
    "valor": 208.99,
  },
  {
    "fecha": "2023-03-11",
    "valor": 209.43,
  },
  {
    "fecha": "2023-03-12",
    "valor": 209.43,
  },
  {
    "fecha": "2023-03-13",
    "valor": 209.43,
  },
  {
    "fecha": "2023-03-14",
    "valor": 210.74,
  },
  {
    "fecha": "2023-03-15",
    "valor": 211.18,
  },
  {
    "fecha": "2023-03-16",
    "valor": 211.62,
  },
  {
    "fecha": "2023-03-17",
    "valor": 212.06,
  },
  {
    "fecha": "2023-03-18",
    "valor": 212.49,
  },
  {
    "fecha": "2023-03-19",
    "valor": 212.49,
  },
  {
    "fecha": "2023-03-20",
    "valor": 212.49,
  },
  {
    "fecha": "2023-03-21",
    "valor": 213.81,
  },
  {
    "fecha": "2023-03-22",
    "valor": 214.25,
  },
  {
    "fecha": "2023-03-23",
    "valor": 214.7,
  },
  {
    "fecha": "2023-03-24",
    "valor": 215.14,
  },
  {
    "fecha": "2023-03-25",
    "valor": 215.14,
  },
  {
    "fecha": "2023-03-26",
    "valor": 215.14,
  },
  {
    "fecha": "2023-03-27",
    "valor": 215.14,
  },
  {
    "fecha": "2023-03-28",
    "valor": 216.92,
  },
  {
    "fecha": "2023-03-29",
    "valor": 217.37,
  },
  {
    "fecha": "2023-03-30",
    "valor": 217.82,
  },
  {
    "fecha": "2023-03-31",
    "valor": 218.27,
  },
  {
    "fecha": "2023-04-01",
    "valor": 218.72,
  },
  {
    "fecha": "2023-04-02",
    "valor": 218.72,
  },
  {
    "fecha": "2023-04-03",
    "valor": 218.72,
  },
  {
    "fecha": "2023-04-04",
    "valor": 220.07,
  },
  {
    "fecha": "2023-04-05",
    "valor": 220.53,
  },
  {
    "fecha": "2023-04-06",
    "valor": 220.98,
  },
  {
    "fecha": "2023-04-07",
    "valor": 220.98,
  },
  {
    "fecha": "2023-04-08",
    "valor": 220.98,
  },
  {
    "fecha": "2023-04-09",
    "valor": 220.98,
  },
  {
    "fecha": "2023-04-10",
    "valor": 220.98,
  },
  {
    "fecha": "2023-04-11",
    "valor": 223.27,
  },
  {
    "fecha": "2023-04-12",
    "valor": 223.73,
  },
  {
    "fecha": "2023-04-13",
    "valor": 224.2,
  },
  {
    "fecha": "2023-04-14",
    "valor": 224.66,
  },
  {
    "fecha": "2023-04-15",
    "valor": 225.12,
  },
  {
    "fecha": "2023-04-16",
    "valor": 225.12,
  },
  {
    "fecha": "2023-04-17",
    "valor": 225.12,
  },
  {
    "fecha": "2023-04-18",
    "valor": 226.7,
  },
  {
    "fecha": "2023-04-19",
    "valor": 227.27,
  },
  {
    "fecha": "2023-04-20",
    "valor": 227.83,
  },
  {
    "fecha": "2023-04-21",
    "valor": 228.39,
  },
  {
    "fecha": "2023-04-22",
    "valor": 228.96,
  },
  {
    "fecha": "2023-04-23",
    "valor": 228.96,
  },
  {
    "fecha": "2023-04-24",
    "valor": 228.96,
  },
  {
    "fecha": "2023-04-25",
    "valor": 230.66,
  },
  {
    "fecha": "2023-04-26",
    "valor": 231.23,
  },
  {
    "fecha": "2023-04-27",
    "valor": 231.81,
  },
  {
    "fecha": "2023-04-28",
    "valor": 232.38,
  },
  {
    "fecha": "2023-04-29",
    "valor": 232.96,
  },
  {
    "fecha": "2023-04-30",
    "valor": 232.96,
  },
  {
    "fecha": "2023-05-01",
    "valor": 232.96,
  },
  {
    "fecha": "2023-05-02",
    "valor": 232.96,
  },
  {
    "fecha": "2023-05-03",
    "valor": 235.27,
  },
  {
    "fecha": "2023-05-04",
    "valor": 235.85,
  },
  {
    "fecha": "2023-05-05",
    "valor": 236.44,
  },
  {
    "fecha": "2023-05-06",
    "valor": 237.02,
  },
  {
    "fecha": "2023-05-07",
    "valor": 237.02,
  },
  {
    "fecha": "2023-05-08",
    "valor": 237.02,
  },
  {
    "fecha": "2023-05-09",
    "valor": 238.79,
  },
  {
    "fecha": "2023-05-10",
    "valor": 239.38,
  },
  {
    "fecha": "2023-05-11",
    "valor": 239.97,
  },
  {
    "fecha": "2023-05-12",
    "valor": 240.57,
  },
  {
    "fecha": "2023-05-13",
    "valor": 241.16,
  },
  {
    "fecha": "2023-05-14",
    "valor": 241.16,
  },
  {
    "fecha": "2023-05-15",
    "valor": 241.16,
  },
  {
    "fecha": "2023-05-16",
    "valor": 242.96,
  },
  {
    "fecha": "2023-05-17",
    "valor": 243.59,
  },
  {
    "fecha": "2023-05-18",
    "valor": 244.22,
  },
  {
    "fecha": "2023-05-19",
    "valor": 244.86,
  },
  {
    "fecha": "2023-05-20",
    "valor": 245.5,
  },
  {
    "fecha": "2023-05-21",
    "valor": 245.5,
  },
  {
    "fecha": "2023-05-22",
    "valor": 245.5,
  },
  {
    "fecha": "2023-05-23",
    "valor": 247.42,
  },
  {
    "fecha": "2023-05-24",
    "valor": 248.07,
  },
  {
    "fecha": "2023-05-25",
    "valor": 248.71,
  },
  {
    "fecha": "2023-05-26",
    "valor": 248.71,
  },
  {
    "fecha": "2023-05-27",
    "valor": 248.71,
  },
  {
    "fecha": "2023-05-28",
    "valor": 248.71,
  },
  {
    "fecha": "2023-05-29",
    "valor": 248.71,
  },
  {
    "fecha": "2023-05-30",
    "valor": 251.97,
  },
  {
    "fecha": "2023-05-31",
    "valor": 252.63,
  },
  {
    "fecha": "2023-06-01",
    "valor": 253.28,
  },
  {
    "fecha": "2023-06-02",
    "valor": 253.94,
  },
  {
    "fecha": "2023-06-03",
    "valor": 254.61,
  },
  {
    "fecha": "2023-06-04",
    "valor": 254.61,
  },
  {
    "fecha": "2023-06-05",
    "valor": 254.61,
  },
  {
    "fecha": "2023-06-06",
    "valor": 256.6,
  },
  {
    "fecha": "2023-06-07",
    "valor": 257.27,
  },
  {
    "fecha": "2023-06-08",
    "valor": 257.94,
  },
  {
    "fecha": "2023-06-09",
    "valor": 258.61,
  },
  {
    "fecha": "2023-06-10",
    "valor": 259.29,
  },
  {
    "fecha": "2023-06-11",
    "valor": 259.29,
  },
  {
    "fecha": "2023-06-12",
    "valor": 259.29,
  },
  {
    "fecha": "2023-06-13",
    "valor": 261.32,
  },
  {
    "fecha": "2023-06-14",
    "valor": 262,
  },
  {
    "fecha": "2023-06-15",
    "valor": 262.68,
  },
  {
    "fecha": "2023-06-16",
    "valor": 263.36,
  },
  {
    "fecha": "2023-06-17",
    "valor": 264.02,
  },
  {
    "fecha": "2023-06-18",
    "valor": 264.02,
  },
  {
    "fecha": "2023-06-19",
    "valor": 264.02,
  },
  {
    "fecha": "2023-06-20",
    "valor": 264.02,
  },
  {
    "fecha": "2023-06-21",
    "valor": 264.02,
  },
  {
    "fecha": "2023-06-22",
    "valor": 267.35,
  },
  {
    "fecha": "2023-06-23",
    "valor": 268.02,
  },
  {
    "fecha": "2023-06-24",
    "valor": 268.69,
  },
  {
    "fecha": "2023-06-25",
    "valor": 268.69,
  },
  {
    "fecha": "2023-06-26",
    "valor": 268.69,
  },
  {
    "fecha": "2023-06-27",
    "valor": 270.72,
  },
  {
    "fecha": "2023-06-28",
    "valor": 271.4,
  },
  {
    "fecha": "2023-06-29",
    "valor": 272.08,
  },
  {
    "fecha": "2023-06-30",
    "valor": 272.76,
  },
  {
    "fecha": "2023-07-01",
    "valor": 273.44,
  },
  {
    "fecha": "2023-07-02",
    "valor": 273.44,
  },
  {
    "fecha": "2023-07-03",
    "valor": 273.44,
  },
  {
    "fecha": "2023-07-04",
    "valor": 275.5,
  },
  {
    "fecha": "2023-07-05",
    "valor": 276.2,
  },
  {
    "fecha": "2023-07-06",
    "valor": 276.89,
  },
  {
    "fecha": "2023-07-07",
    "valor": 277.58,
  },
  {
    "fecha": "2023-07-08",
    "valor": 278.28,
  },
  {
    "fecha": "2023-07-09",
    "valor": 278.28,
  },
  {
    "fecha": "2023-07-10",
    "valor": 278.28,
  },
  {
    "fecha": "2023-07-11",
    "valor": 280.38,
  },
  {
    "fecha": "2023-07-12",
    "valor": 281.08,
  },
  {
    "fecha": "2023-07-13",
    "valor": 281.78,
  },
  {
    "fecha": "2023-07-14",
    "valor": 282.49,
  },
  {
    "fecha": "2023-07-15",
    "valor": 283.2,
  },
  {
    "fecha": "2023-07-16",
    "valor": 283.2,
  },
  {
    "fecha": "2023-07-17",
    "valor": 283.2,
  },
  {
    "fecha": "2023-07-18",
    "valor": 284.98,
  },
  {
    "fecha": "2023-07-19",
    "valor": 285.51,
  },
  {
    "fecha": "2023-07-20",
    "valor": 286.05,
  },
  {
    "fecha": "2023-07-21",
    "valor": 286.59,
  },
  {
    "fecha": "2023-07-22",
    "valor": 287.13,
  },
  {
    "fecha": "2023-07-23",
    "valor": 287.13,
  },
  {
    "fecha": "2023-07-24",
    "valor": 287.13,
  },
  {
    "fecha": "2023-07-25",
    "valor": 288.75,
  },
  {
    "fecha": "2023-07-26",
    "valor": 289.29,
  },
  {
    "fecha": "2023-07-27",
    "valor": 289.84,
  },
  {
    "fecha": "2023-07-28",
    "valor": 290.38,
  },
  {
    "fecha": "2023-07-29",
    "valor": 290.93,
  },
  {
    "fecha": "2023-07-30",
    "valor": 290.93,
  },
  {
    "fecha": "2023-07-31",
    "valor": 290.93,
  },
  {
    "fecha": "2023-08-01",
    "valor": 292.58,
  },
  {
    "fecha": "2023-08-02",
    "valor": 293.13,
  },
  {
    "fecha": "2023-08-03",
    "valor": 293.68,
  },
  {
    "fecha": "2023-08-04",
    "valor": 294.23,
  },
  {
    "fecha": "2023-08-05",
    "valor": 294.78,
  },
  {
    "fecha": "2023-08-06",
    "valor": 294.78,
  },
  {
    "fecha": "2023-08-07",
    "valor": 294.78,
  },
  {
    "fecha": "2023-08-08",
    "valor": 296.45,
  },
  {
    "fecha": "2023-08-09",
    "valor": 297.01,
  },
  {
    "fecha": "2023-08-10",
    "valor": 297.57,
  },
  {
    "fecha": "2023-08-11",
    "valor": 298.13,
  },
  {
    "fecha": "2023-08-12",
    "valor": 298.69,
  },
  {
    "fecha": "2023-08-13",
    "valor": 298.69,
  },
  {
    "fecha": "2023-08-14",
    "valor": 298.69,
  },
  {
    "fecha": "2023-08-15",
    "valor": 300.38,
  },
  {
    "fecha": "2023-08-16",
    "valor": 300.94,
  },
  {
    "fecha": "2023-08-17",
    "valor": 301.54,
  },
  {
    "fecha": "2023-08-18",
    "valor": 302.13,
  },
  {
    "fecha": "2023-08-19",
    "valor": 302.73,
  },
  {
    "fecha": "2023-08-20",
    "valor": 302.73,
  },
  {
    "fecha": "2023-08-21",
    "valor": 302.73,
  },
  {
    "fecha": "2023-08-22",
    "valor": 302.73,
  },
  {
    "fecha": "2023-08-23",
    "valor": 305.12,
  },
  {
    "fecha": "2023-08-24",
    "valor": 305.72,
  },
  {
    "fecha": "2023-08-25",
    "valor": 306.33,
  },
  {
    "fecha": "2023-08-26",
    "valor": 306.93,
  },
  {
    "fecha": "2023-08-27",
    "valor": 306.93,
  },
  {
    "fecha": "2023-08-28",
    "valor": 306.93,
  },
  {
    "fecha": "2023-08-29",
    "valor": 308.75,
  },
  {
    "fecha": "2023-08-30",
    "valor": 309.36,
  },
  {
    "fecha": "2023-08-31",
    "valor": 309.97,
  },
  {
    "fecha": "2023-09-01",
    "valor": 310.58,
  },
  {
    "fecha": "2023-09-02",
    "valor": 311.2,
  },
  {
    "fecha": "2023-09-03",
    "valor": 311.2,
  },
  {
    "fecha": "2023-09-04",
    "valor": 311.2,
  },
  {
    "fecha": "2023-09-05",
    "valor": 313.04,
  },
  {
    "fecha": "2023-09-06",
    "valor": 313.66,
  },
  {
    "fecha": "2023-09-07",
    "valor": 314.28,
  },
  {
    "fecha": "2023-09-08",
    "valor": 314.9,
  },
  {
    "fecha": "2023-09-09",
    "valor": 315.52,
  },
  {
    "fecha": "2023-09-10",
    "valor": 315.52,
  },
  {
    "fecha": "2023-09-11",
    "valor": 315.52,
  },
  {
    "fecha": "2023-09-12",
    "valor": 317.39,
  },
  {
    "fecha": "2023-09-13",
    "valor": 318.02,
  },
  {
    "fecha": "2023-09-14",
    "valor": 318.64,
  },
  {
    "fecha": "2023-09-15",
    "valor": 319.27,
  },
  {
    "fecha": "2023-09-16",
    "valor": 319.9,
  },
  {
    "fecha": "2023-09-17",
    "valor": 319.9,
  },
  {
    "fecha": "2023-09-18",
    "valor": 319.9,
  },
  {
    "fecha": "2023-09-19",
    "valor": 323.66,
  },
  {
    "fecha": "2023-09-20",
    "valor": 324.93,
  },
  {
    "fecha": "2023-09-21",
    "valor": 326.19,
  },
  {
    "fecha": "2023-09-22",
    "valor": 327.47,
  },
  {
    "fecha": "2023-09-23",
    "valor": 328.75,
  },
  {
    "fecha": "2023-09-24",
    "valor": 328.75,
  },
  {
    "fecha": "2023-09-25",
    "valor": 328.75,
  },
  {
    "fecha": "2023-09-26",
    "valor": 332.61,
  },
  {
    "fecha": "2023-09-27",
    "valor": 333.91,
  },
  {
    "fecha": "2023-09-28",
    "valor": 335.21,
  },
  {
    "fecha": "2023-09-29",
    "valor": 336.52,
  },
  {
    "fecha": "2023-09-30",
    "valor": 337.84,
  },
  {
    "fecha": "2023-10-01",
    "valor": 337.84,
  },
  {
    "fecha": "2023-10-02",
    "valor": 337.84,
  },
  {
    "fecha": "2023-10-03",
    "valor": 341.81,
  },
  {
    "fecha": "2023-10-04",
    "valor": 343.14,
  },
  {
    "fecha": "2023-10-05",
    "valor": 344.48,
  },
  {
    "fecha": "2023-10-06",
    "valor": 345.83,
  },
  {
    "fecha": "2023-10-07",
    "valor": 347.18,
  },
  {
    "fecha": "2023-10-08",
    "valor": 347.18,
  },
  {
    "fecha": "2023-10-09",
    "valor": 347.18,
  },
  {
    "fecha": "2023-10-10",
    "valor": 351.26,
  },
  {
    "fecha": "2023-10-11",
    "valor": 352.63,
  },
  {
    "fecha": "2023-10-12",
    "valor": 354.01,
  },
  {
    "fecha": "2023-10-13",
    "valor": 355.39,
  },
  {
    "fecha": "2023-10-14",
    "valor": 355.39,
  },
  {
    "fecha": "2023-10-15",
    "valor": 355.39,
  },
  {
    "fecha": "2023-10-16",
    "valor": 355.39,
  },
  {
    "fecha": "2023-10-17",
    "valor": 355.39,
  },
  {
    "fecha": "2023-10-18",
    "valor": 362.35,
  },
  {
    "fecha": "2023-10-19",
    "valor": 363.75,
  },
  {
    "fecha": "2023-10-20",
    "valor": 365.16,
  },
  {
    "fecha": "2023-10-21",
    "valor": 366.57,
  },
  {
    "fecha": "2023-10-22",
    "valor": 366.57,
  },
  {
    "fecha": "2023-10-23",
    "valor": 366.57,
  },
  {
    "fecha": "2023-10-24",
    "valor": 370.84,
  },
  {
    "fecha": "2023-10-25",
    "valor": 372.27,
  },
  {
    "fecha": "2023-10-26",
    "valor": 373.71,
  },
  {
    "fecha": "2023-10-27",
    "valor": 375.15,
  },
  {
    "fecha": "2023-10-28",
    "valor": 376.6,
  },
  {
    "fecha": "2023-10-29",
    "valor": 376.6,
  },
  {
    "fecha": "2023-10-30",
    "valor": 376.6,
  },
  {
    "fecha": "2023-10-31",
    "valor": 380.98,
  },
  {
    "fecha": "2023-11-01",
    "valor": 382.46,
  },
  {
    "fecha": "2023-11-02",
    "valor": 383.93,
  },
  {
    "fecha": "2023-11-03",
    "valor": 385.42,
  },
  {
    "fecha": "2023-11-04",
    "valor": 386.91,
  },
  {
    "fecha": "2023-11-05",
    "valor": 386.91,
  },
  {
    "fecha": "2023-11-06",
    "valor": 386.91,
  },
  {
    "fecha": "2023-11-07",
    "valor": 386.91,
  },
  {
    "fecha": "2023-11-08",
    "valor": 392.92,
  },
  {
    "fecha": "2023-11-09",
    "valor": 394.44,
  },
  {
    "fecha": "2023-11-10",
    "valor": 395.96,
  },
  {
    "fecha": "2023-11-11",
    "valor": 397.49,
  },
  {
    "fecha": "2023-11-12",
    "valor": 397.49,
  },
  {
    "fecha": "2023-11-13",
    "valor": 397.49,
  },
  {
    "fecha": "2023-11-14",
    "valor": 402.12,
  },
  {
    "fecha": "2023-11-15",
    "valor": 403.67,
  },
  {
    "fecha": "2023-11-16",
    "valor": 405.23,
  },
  {
    "fecha": "2023-11-17",
    "valor": 406.31,
  },
  {
    "fecha": "2023-11-18",
    "valor": 407.39,
  },
  {
    "fecha": "2023-11-19",
    "valor": 407.39,
  },
  {
    "fecha": "2023-11-20",
    "valor": 407.39,
  },
  {
    "fecha": "2023-11-21",
    "valor": 407.39,
  },
  {
    "fecha": "2023-11-22",
    "valor": 411.75,
  },
  {
    "fecha": "2023-11-23",
    "valor": 412.84,
  },
  {
    "fecha": "2023-11-24",
    "valor": 413.94,
  },
  {
    "fecha": "2023-11-25",
    "valor": 415.04,
  },
  {
    "fecha": "2023-11-26",
    "valor": 415.04,
  },
  {
    "fecha": "2023-11-27",
    "valor": 415.04,
  },
  {
    "fecha": "2023-11-28",
    "valor": 418.37,
  },
  {
    "fecha": "2023-11-29",
    "valor": 419.48,
  },
  {
    "fecha": "2023-11-30",
    "valor": 420.6,
  },
  {
    "fecha": "2023-12-01",
    "valor": 421.72,
  },
  {
    "fecha": "2023-12-02",
    "valor": 422.84,
  },
  {
    "fecha": "2023-12-03",
    "valor": 422.84,
  },
  {
    "fecha": "2023-12-04",
    "valor": 422.84,
  },
  {
    "fecha": "2023-12-05",
    "valor": 426.22,
  },
  {
    "fecha": "2023-12-06",
    "valor": 427.36,
  },
  {
    "fecha": "2023-12-07",
    "valor": 428.49,
  },
  {
    "fecha": "2023-12-08",
    "valor": 429.64,
  },
  {
    "fecha": "2023-12-09",
    "valor": 429.64,
  },
  {
    "fecha": "2023-12-10",
    "valor": 429.64,
  },
  {
    "fecha": "2023-12-11",
    "valor": 429.64,
  },
  {
    "fecha": "2023-12-12",
    "valor": 434.23,
  },
  {
    "fecha": "2023-12-13",
    "valor": 435.38,
  },
  {
    "fecha": "2023-12-14",
    "valor": 436.54,
  },
  {
    "fecha": "2023-12-15",
    "valor": 437.7,
  },
  {
    "fecha": "2023-12-16",
    "valor": 438.87,
  },
  {
    "fecha": "2023-12-17",
    "valor": 438.87,
  },
  {
    "fecha": "2023-12-18",
    "valor": 438.87,
  },
  {
    "fecha": "2023-12-19",
    "valor": 444.01,
  },
  {
    "fecha": "2023-12-20",
    "valor": 445.74,
  },
  {
    "fecha": "2023-12-21",
    "valor": 447.48,
  },
  {
    "fecha": "2023-12-22",
    "valor": 449.22,
  },
  {
    "fecha": "2023-12-23",
    "valor": 450.97,
  },
  {
    "fecha": "2023-12-24",
    "valor": 450.97,
  },
  {
    "fecha": "2023-12-25",
    "valor": 450.97,
  },
  {
    "fecha": "2023-12-26",
    "valor": 450.97,
  },
  {
    "fecha": "2023-12-27",
    "valor": 458.03,
  },
  {
    "fecha": "2023-12-28",
    "valor": 459.81,
  },
  {
    "fecha": "2023-12-29",
    "valor": 461.6,
  },
  {
    "fecha": "2023-12-30",
    "valor": 463.4,
  },
  {
    "fecha": "2023-12-31",
    "valor": 463.4,
  },
  {
    "fecha": "2024-01-01",
    "valor": 463.4,
  },
]
`;
